sklearn/covariance/_shrunk_covariance.py

```diff
@@ -535,17 +535,19 @@ def oas(X, *, assume_centered=False):
     else:
         n_samples, n_features = X.shape
 
-    emp_cov = empirical_covariance(X, assume_centered=assume_centered)
-    mu = np.trace(emp_cov) / n_features
-
+    # Resolves differences with OAS when ussing uncentered data
+    if not assume_centered:
+        X = X - X.mean(0)
+    emp_cov = empirical_covariance(X, assume_centered=True)
+    mu = np.trace(emp_cov)
     # formula from Chen et al.'s **implementation**
-    alpha = np.mean(emp_cov**2)
+    alpha = np.trace(emp_cov**2)
```

Comment:
@GaelVaroquaux Wondering if you'd be able to review this PR. It's comprised of 4 relatively minor changes. This one being the most important.

Reply:
I don't really get why we were taking the mean instead of the trace there. It does not look like what is written and proven in the paper. I agree with the change.